ACE/ace/Malloc_T.h

```diff
@@ -203,6 +207,101 @@ class ACE_Dynamic_Cached_Allocator : public ACE_New_Allocator
   size_t chunk_size_;
 };
 
+/**
+ * @class ACE_Cascaded_Dynamic_Cached_Allocator
+ *
+ * @brief A fixed-size-based allocator that caches blocks for quicker access,
+ * but with a hierarchy of cascaded, nested allocators
+ *
+ * This class enables caching of dynamically allocated,
+ * fixed-size chunks. Notice that the @a chunk_size
+ * must be greater than or equal to <code> sizeof (void*) </code> for
+ * this to work properly.
+ *
+ * Notice that when the latest allocator is empty, the allocator will create a fresh
+ * @a ACE_Dynamic_Cached_Allocator allocator again with
+ * <code> init_n_chunks* the sum of current allocators </code> as it's constructor parameter,
+ * so all the allocators will form a cascaded hierarchy.
+
+ * This class can be configured flexibly with different types of
+ * ACE_LOCK strategies that support the @a ACE_Thread_Mutex and @a
+ * ACE_Process_Mutex constructor API.
+ *
+ * @sa ACE_Dynamic_Cached_Allocator
+ */
+template <class ACE_LOCK>
+class ACE_Cascaded_Dynamic_Cached_Allocator : public ACE_Allocator
+{
+public:
+  /// Create a cached memory pool with @a n_chunks chunks
+  /// each with @a chunk_size size.
+  ACE_Cascaded_Dynamic_Cached_Allocator (size_t initial_n_chunks, size_t chunk_size);
+
+  /// Clear things up.
+  ~ACE_Cascaded_Dynamic_Cached_Allocator ();
+
+  /**
+   * Get a chunk of memory from free list cache.  Note that @a nbytes is
+   * only checked to make sure that it's less or equal to @a chunk_size,
+   * and is otherwise ignored since malloc() always returns a pointer to an
+   * item of @a chunk_size size.
+   */
+  virtual void *malloc (size_t nbytes);
+
+  /**
+   * Get a chunk of memory from free list cache, giving them
+   * @a initial_value.  Note that @a nbytes is only checked to make sure
+   * that it's less or equal to @a chunk_size, and is otherwise ignored
+   * since calloc() always returns a pointer to an item of @a chunk_size.
+   */
+  virtual void *calloc (size_t nbytes, char initial_value = '\0');
+
+  /// This method is a no-op and just returns 0 since the free list
+  /// only works with fixed sized entities.
+  virtual void *calloc (size_t n_elem, size_t elem_size, char initial_value = '\0');
+
+  /// Return a chunk of memory back to free list cache.
+  virtual void free (void *ptr);
+
+  /// These methods are no-ops.
+  virtual int remove ();
```

Comment:
override instead of virtual?

Reply:
my base class is Allocator, so no implemention can be overrided

By the way I copy the method declarations from `ACE_Static_Allocator_Base` :(

Should these be overrided ?